packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -490,6 +564,15 @@ export const GridRootStyles = styled('div', {
         display: 'flex',
       },
     },
+    [`& .${gridClasses['row--lastVisible']}`]: {
+      minHeight: 'unset !important',
+      maxHeight: 'unset !important',
+    },
+    [`& .${gridClasses['row--lastVisible']} .${gridClasses.cell}`]: {
+      width: 'var(--width)',
+      height: 'calc(var(--height) + var(--DataGrid-hasScrollX) * var(--DataGrid-scrollbarSize))',
+      paddingBottom: 'calc(var(--DataGrid-hasScrollX) * var(--DataGrid-scrollbarSize))',
```

Comment:
Why should the last row have different bottom padding?
https://deploy-preview-10059--material-ui-x.netlify.app/x/react-data-grid/demo/

<img width="1020" src="https://github.com/mui/mui-x/assets/13808724/f4ba8633-a0e8-44e1-8978-a3fa6d861e6a">

Reply:
Initially, there was the logic to add padding to the last column so it stretched to occupy the space of the vertical scrollbar:

![image](https://github.com/mui/mui-x/assets/1423607/ac410f7b-1875-4bed-aab7-73f8aba127b7)

So when I added the fake vertical scrollbars, the cells got the same padding to fill the space they now need to fill, because the scrollbar floats in front of the whole grid. Then when I added the horizontal scrollbar, I used the same logic to space thing.

I could look for other options to fill the space if there are unsolvable problems with that approach.

Reply:
> So when I added the fake vertical scrollbars, the cells got the same padding to fill the space they now need to fill, because the scrollbar floats in front of the whole grid.

Does this mean, that the last row should always have the bottom padding?

Reply:
Yeah right now it needs bottom padding.

One alternative could be that the bottom scrollbar space is filled with a filler div instead. But I'd keep the right scrollbar space filled with padding, because it's not possible to fill it cleanly with a single div (there would need to be one div inside each row that needs padding).

Reply:
A filler div for the horizontal scroll bar sounds like a better option to me!
This would also make the bottom cell outline in the last row visible:

<img width="859" src="https://github.com/mui/mui-x/assets/13808724/4c42e49d-164c-4e13-8173-852d7fa26f02">

It's a minor thing, but still 🙂

Reply:
Ok this is fixed with the spacers, which is indeed better, but wow this took all day and I have a renewed hate for our styling solution and for CSS in general.